src/sympc/protocol/aby3/aby3.py

```diff
@@ -0,0 +1,144 @@
+            return False
+
+        return True
+
+    @staticmethod
+    def trunc1(
+        ptr_list: List["ReplicatedSharedTensor"], shape: torch.Size, session: Session
+    ) -> List["ReplicatedSharedTensor"]:
+        """Performs the ABY3 trunc1 algorithm.
+
+        Args:
+            ptr_list (List[ReplicatedSharedTensor]): Tensors to truncate
+            shape(torch.Size) : shape of tensor values
+            session(Session) : session the tensor belong to
+
+        Returns:
+            List["ReplicatedSharedTensor"] : Truncated shares.
+        """
+        gen = csprng.create_random_device_generator()
+        rand_value = torch.empty(size=shape, dtype=session.tensor_type).random_(
+            generator=gen
+        )
+        base = session.config.encoder_base
+        precision = session.config.encoder_precision
+        scale = base ** precision
+        x1 = ptr_list[0].get_copy().shares[0]
+        x2, x3 = ptr_list[1].get_copy().shares
+        x1_trunc = x1 >> precision if base == 2 else x1 // scale
+        x_trunc = (x2 + x3) >> precision if base == 2 else (x2 + x3) // scale
+        shares = [x1_trunc, x_trunc - rand_value, rand_value]
+        ptr_list = ReplicatedSharedTensor.distribute_shares(shares, session)
+        return ptr_list
+
+    @staticmethod
+    def getTruncationPair(x: MPCTensor, session: Session) -> Tuple[MPCTensor]:
+        """Generates truncation pair for the given MPCTensor.
+
+        Args:
+            x (MPCTensor): input tensor
+            session (Session) : session of the input tensor.
+
+        Returns:
+            Tuple[MPCTensor]: generated truncation pair.
+        """
+        r: List = []
+        rPrime: List = []
+        for session_ptrs in session.session_ptrs:
+            rst = session_ptrs.prrs_generate_random_share(shape=x.shape)
+            rst = rst.resolve_pointer_type()
+            rPrime.append(rst)
+            r.append(rst)
+
+        r = ABY3.trunc1(r, x.shape, session)
+        r_mpc = MPCTensor(shares=r, session=session, shape=x.shape)
+        rPrime_mpc = MPCTensor(shares=rPrime, session=session, shape=x.shape)
+        return r_mpc, rPrime_mpc
+
+    @staticmethod
+    def truncate(x: MPCTensor, session: Session) -> MPCTensor:
+        """Truncates the MPCTensor by scale factor using trunc2 algorithm.
+
+        Args:
+            x (MPCTensor): input tensor
+            session (Session) : session of the input tensor.
+
+        Returns:
+            MPCTensor: truncated MPCTensor.
+
+        Raises:
+            ValueError : parties involved in the computation is not equal to three.
+        """
+        if session.nr_parties != 3:
+            raise ValueError("Share trunc1 algorithm works only for 3 parites.")
+
+        share_ptrs = ABY3.trunc1(x.share_ptrs, x.shape, session)
+        result = MPCTensor(shares=share_ptrs, session=session, shape=x.shape)
+
+        # TODO below trunc2 algorithm erroneous, to be optimized.
```

Comment:
You can keep this truncation algorithm as a function at the end of the file with a note that implementation isn't accurate.